mondrian/src/main/java/mondrian/spi/Dialect.java

```diff
@@ -161,6 +161,18 @@ String quoteIdentifier(
         String name);
 
     /**
+     * Encloses an identifier in quotation marks appropriate for the
+     * current SQL dialect. For example, in Oracle, where the identifiers
+     * are quoted using double-quotes,
+     * <code>quoteIdentifier("schema.name")</code> yields a string
+     * containing <code>"schema"."name"</code>.
+
+     * @param name Name to be quoted.
+     * @return Quoted identifier
+     */
+    String quoteIdentifierWithInnerDots(String name);
```

Comment:
@lucboudreau yes, there is: we need to use the logic from `quoteIdentifierWithInnerDots` ONLY for schema name, and not for the others names (like table name or table alias).
To fully understand this, you should look at the https://github.com/pentaho/mondrian/pull/865/files#diff-ea35a8727d98201e0883f976f9cf6b34R808 and http://jira.pentaho.com/browse/MONDRIAN-2471